crates/swc_ecma_transforms_compat/src/es2015/arrow.rs

```diff
@@ -94,56 +95,9 @@ impl VisitMut for Arrow {
                     })
                     .collect();
 
-                let mut arguments_var = None;
-
-                let in_arrow = self.in_arrow;
-                self.in_arrow = true;
                 body.visit_mut_with(self);
-                self.in_arrow = in_arrow;
-
-                let body = {
-                    let mut arguments_replacer = ArgumentsReplacer {
-                        arguments: private_ident!("_arguments"),
-                        found: false,
-                    };
-                    body.visit_mut_with(&mut arguments_replacer);
-
-                    if arguments_replacer.found {
-                        arguments_var = Some(arguments_replacer.arguments.clone());
-                    }
-                    body
-                };
 
-                let used_this = contains_this_expr(&*body);
-
-                let this_id = if used_this {
-                    Some(private_ident!("_this"))
-                } else {
-                    None
-                };
-
-                if let Some(this_id) = this_id.clone() {
-                    self.vars.push(VarDeclarator {
-                        span: DUMMY_SP,
-                        name: Pat::Ident(this_id.clone().into()),
-                        init: Some(Box::new(Expr::This(ThisExpr { span: DUMMY_SP }))),
-                        definite: false,
-                    });
-
-                    body.visit_mut_with(&mut ThisReplacer { to: &this_id })
-                }
-
-                if let Some(id) = arguments_var.clone() {
-                    self.vars.push(VarDeclarator {
-                        span: DUMMY_SP,
-                        name: Pat::Ident(id.into()),
-                        init: Some(Box::new(Expr::Ident(Ident::new(
-                            js_word!("arguments"),
-                            DUMMY_SP,
-                        )))),
-                        definite: false,
-                    });
-                }
+                body.visit_mut_with(&mut FunctionWrapper::new(&mut self.state));
```

Comment:
This has exponential time complexity, but it's fine for most case because `n` is super small. I think it's better to have a comment mentioning it.